src/Microsoft.TestPlatform.ObjectModel/Client/Events/DiscoveredSource.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+
+#nullable disable
+
+namespace Microsoft.VisualStudio.TestPlatform.ObjectModel.Client;
+
+/// <summary>
+/// Describes a source file that was discovered, or that was requested to be discovered.
+/// </summary>
+public class DiscoveredSource
```

Comment:
I was actually thinking about splitting it to have a discovery specific object for transport (*Payload), and internal object for a source, or even Result. But I don't want to overthink this, because in ideal world the payload would be public but the result would not, and so we can hopefully split them later if needed.